Makefile

```diff
@@ -236,6 +241,9 @@ verify-tiltfile: ## Verify Tiltfile format.
 
 ##@ Development:
 
+.PHONY: install-tools # populate hack/tools/bin
+install-tools: $(ENVSUBST) $(KUSTOMIZE) $(KUBECTL) $(HELM) $(GINKGO)
```

Comment:
Unifying these installs together made it a lot easier to standardize dep installs across the various jobs and scripts.

Should I go ahead and add the remaining set of tools in hack/tools/bin to this list for best practice going forward (despite the fact that not every invocation of `make install-tools` will require *every single* binary)?

cc @CecileRobertMichon @jsturtevant

Reply:
> despite the fact that not every invocation of make install-tools will require every single binary

I'm weak -1 on this, even if it improves the makefile a little, it would make running, say `make go-test`, a lot slower the first time and install a bunch of binaries that aren't needed, I slightly prefer the "just in time" install approach.

Reply:
Cool, yeah, that's the tradeoff.

Any concerns w/ this new `install-tools` target here? The practical problem is we have > 1 (actually like 3 or 4) places in > 1 file that need the above (or at least 4 of 5) installed, and during this effort I encountered how easy it is to miss a dep when adding a new one if the only option is to copy/paste those individual tool install commands around.

Reply:
I'm also mildly opposed to this change if it adds any noticeable time to testing targets. Could we maybe have the new target install just the tools that are actually shared, or is this still a maintenance headache?

```Makefile
install-tools: $(ENVSUBST) $(KUSTOMIZE) $(KUBECTL)

test-e2e-run: generate-e2e-templates install-tools $(GINKGO)
```

Reply:
only `test-e2e-run` and `tilt-up` have been updated to include this new target, so I think that's a good trade-off (those are flows that are already sort of heavyweight and shouldn't slow down CI overall)